### PR TITLE
Implement additional archive types

### PR DESCRIPTION
**Pull request by Diggers89**
Implementations of the following:
- Rar
- Tar
- 7Zip
- GZip

### CODE REVIEW

File: Source/Service/ArchiveProcessor.cs [hunk 1]
```diff
@@ -96,7 +96,7 @@ private Task ProcessArchive()
             _logger.LogInformation($"File Id: {_config.ArchiveFileId} Extracting archive to temp folder {_tmpOriginalDirectory}");
             var fileMappings = _archiveManager.ExtractArchive(_config.InputPath, _tmpOriginalDirectory);
 
-            if (fileMappings == null)
+            if (fileMappings.Count == 0)
```
Comment (BigPeeg) [inline Source/Service/ArchiveProcessor.cs:99]: What is the outcome if an empty archive file is processed?

File: Source/Service/Archive/TarArchiveManager.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                        var fileId = Guid.NewGuid();
+                        fileMapping.Add(fileId, entry.Key);
+                        entry.WriteToFile($"{targetPath}/{fileId}");
+                    }
+                }
+
+            }
+            catch(Exception e)
+            {
+                _logger.LogError($"Archive File Path: {archiveFilePath}, error extracting archive. {e.Message}");
```
Comment (BigPeeg) [inline Source/Service/Archive/TarArchiveManager.cs:64]: If this exception is thrown, then not all of the archive contents will be unpacked, which will mean that only a subset of the files will be processed. If this occurs, then we should be adding an error report saying that now all the files were processed.

**Review by BigPeeg — APPROVED** [review]
I'm happy for the partial extract comment to be handled as a bug and sorted out later.

File: Source/Service/ArchiveProcessor.cs (hunk 1 above)
Comment (Diggers89) [reply Source/Service/ArchiveProcessor.cs]: Removed now in favor of the FileEncryptedException

File: Source/Service/Archive/TarArchiveManager.cs (hunk 2 above)
Comment (Diggers89) [reply Source/Service/Archive/TarArchiveManager.cs:66]: #19

**Pull request merged**